internal/environment/environment.go

```diff
@@ -361,3 +336,29 @@ func (n *NomadEnvironment) removeRunners(apiClient nomad.ExecutorAPI, count uint
 	}
 	return nil
 }
+
+func (n *NomadEnvironment) removeRunners(apiClient nomad.ExecutorAPI) error {
+	// Only to avoid timing issues as an idle runner is also removed when Nomad has deleted the allocation.
+	for _, r := range n.idleRunners.List() {
+		n.idleRunners.Delete(r.ID())
+	}
```

Comment:
Could you please add a sentence _what_ these lines are intended to do (in addition to _why_ they are included)? If I got it right, I would state something like:

```suggestion
        // Purge the list of available idle runners for the given execution environment // maybe own method in storage.go?
        // This prevents a race condition where an idle runner is returned while removing the environment
	for _, r := range n.idleRunners.List() {
		n.idleRunners.Delete(r.ID())
	}
```

Reply:
I have added a description to the new `Purge` function

Reply:
Ah, thank you! My assumption was not correct but now I get the cause with your extended comment 🙂